enterprise/app/services/enterprise/billing/handle_stripe_event_service.rb

```diff
@@ -73,6 +77,26 @@ def process_subscription_deleted
     Enterprise::Billing::CreateStripeCustomerService.new(account: account).perform
   end
 
+  def process_checkout_session_completed
+    session = @event.data.object
+    metadata = session.metadata
+
+    # Only process topup checkout sessions
+    return unless metadata&.[]('topup') == 'true'
+
+    topup_account = Account.find_by(id: metadata['account_id'])
+    return if topup_account.blank?
+
+    credits = metadata['credits'].to_i
+    amount_cents = metadata['amount_cents'].to_i
+    currency = metadata['currency'] || 'usd'
+    Enterprise::Billing::TopupFulfillmentService.new(account: topup_account).fulfill(
+      credits: credits,
+      amount_cents: amount_cents,
+      currency: currency
+    )
+  end
```

Comment:
Let's catch any error here instead of the `fulfill` method. And track the exception here, and include any details we have from stripe as well. Maybe using Rails.logger.error. That way we have enough data to fix any data after the fact